src/Microsoft.AspNetCore.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -65,7 +64,7 @@ public DateHeaderValues GetDateHeaderValues()
 
             return new DateHeaderValues()
             {
-                Bytes = _activeDateBytes ? _dateBytes0 : _dateBytes1,
+                Bytes = _dateBytes,
```

Comment:
Previously _activeDateBytes was volatile, so there was a volatile read prior to reading _dateBytes0 or _dateBytes1 and a volatile write after updating them.  Does _dateBytes need to be volatile?